### PR TITLE
Feature/86cwh975e buffer for requests

### PR DESCRIPTION
**Pull request by NikkiShao**
- added a loading after each submit/save button is pressed so it cant be spammed
- also added in some confirmation/error messages that were missing/inconsistent before

for: 
- account settings pages
- add/edit post
- add/edit service
- bookings (after entering payment details)

### CODE REVIEW

File: behind-the-veil-siteroot/imports/ui/components/pages/profile/settings/AccountDetails.jsx [hunk 1]
```diff
@@ -139,25 +141,52 @@ export const AccountDetails = () => {
         // Update text errors state with new error messages
         setTextErrors(newErrors);
 
+        console.log(isError)
```
Comment (joshualoongwy) [inline behind-the-veil-siteroot/imports/ui/components/pages/profile/settings/AccountDetails.jsx:144]: is this console log still needed

File: behind-the-veil-siteroot/imports/ui/components/pages/profile/settings/AccountDetails.jsx [hunk 2]
```diff
@@ -139,25 +141,52 @@ export const AccountDetails = () => {
...
+                        (error) => {
+                            if (error) {
+                                reject(error)
+                            }
+                        });
+                }
+                
+                // update profile image
+               if (imageObject !== userInfo.profileImage) {
+                    console.log(imageObject)
```
Comment (joshualoongwy) [inline behind-the-veil-siteroot/imports/ui/components/pages/profile/settings/AccountDetails.jsx:170]: log

File: behind-the-veil-siteroot/imports/ui/components/pages/artist/AddEditPostPage.jsx [hunk 3]
```diff
@@ -190,8 +198,12 @@ export const AddEditPostPage = ({ isEdit }) => {
           }
         );
       }
-    })
-      .catch((reason) => alert(reason));
+    }).then(() => {
```
Comment (joshualoongwy) [inline behind-the-veil-siteroot/imports/ui/components/pages/artist/AddEditPostPage.jsx:201]: Just for future reference, I think you can avoid duplicating logic (i.e. `setIsSubmitting(false)`) in .then() and .catch() by including the logic in .finally()
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

but if it's working rn then just leave it

File: behind-the-veil-siteroot/imports/ui/components/pages/profile/settings/AccountDetails.jsx (hunk 1 above)
Comment (NikkiShao) [reply behind-the-veil-siteroot/imports/ui/components/pages/profile/settings/AccountDetails.jsx:144]: removed

File: behind-the-veil-siteroot/imports/ui/components/pages/profile/settings/AccountDetails.jsx (hunk 2 above)
Comment (NikkiShao) [reply behind-the-veil-siteroot/imports/ui/components/pages/profile/settings/AccountDetails.jsx:170]: removed

File: behind-the-veil-siteroot/imports/ui/components/pages/artist/AddEditPostPage.jsx (hunk 3 above)
Comment (NikkiShao) [reply behind-the-veil-siteroot/imports/ui/components/pages/artist/AddEditPostPage.jsx:201]: okiii noted!

**Review by joshualoongwy — APPROVED** [review]

**Pull request merged**